src/common.rs

```diff
@@ -16,9 +16,9 @@ pub enum ColorType {
 
 impl ColorType {
     /// Returns the number of samples used per pixel of `ColorType`
-    pub fn samples(&self) -> usize {
+    pub fn samples(self) -> usize {
```

Comment:
This changes the `samples`' argument from `&self` to `self`. I'm not sure it's reasonable.

Reply:
This is probably fine. I assume most users call this function via `c.samples()` rather than `ColorType::samples(&c)` so I wouldn't expect much/any breakage.

Reply:
We could always check instead of guess. Get the larger reverse dependencies [from crates.io's list](https://crates.io/crates/png/reverse_dependencies) and try to compile them with the changes.

Reply:
Those that actually depend on `0.15` anyways

Reply:
We should check manually?

Reply:
It was just an idea to be extra cautious. I don't expect any larger breakage either, and it's certainly easy to fix, so this is optional.

Reply:
Would be cool if there was a tool for automating this though ;)

Reply:
Okay, let's merge this and see whether anything happens.